concordium-consensus/src/Concordium/KonsensusV1/TreeState/StartUp.hs

```diff
@@ -316,7 +322,12 @@ loadCertifiedBlocks ::
     m ()
 loadCertifiedBlocks = do
     certBlocks <- LowLevel.lookupCertifiedBlocks
-    mapM_ loadCertBlock certBlocks
+    -- The first certified block will have the empty parent difference map reference.
+    emptyParent <- DiffMap.newEmptyReference
+    -- Load all certified blocks
+    -- This sets the skov state, puts transactions in the transaction table,
+    -- and reconstructs the account map difference maps for the certified blocks.
+    foldM_ (flip loadCertBlock) emptyParent certBlocks
```

Comment:
This is not correct, because the certified blocks do not in general form a single chain - there may be branches. I think when you fold the accumulator should take a map from block hashes to the DiffMap for the block. Then `loadCertBlock` will look up the DiffMap for the parent, and if it's not found use the `emptyParent` (as the parent is presumably finalized). It will also add the DiffMap constructed for the block to the map.

Reply:
great catch